### PR TITLE
Chore: Fix some tests that break when whatwg-fetch polyfill is imported

### PR DESCRIPTION
**Pull request by tomratcliffe**
## What is this feature?
Fixes some tests that break when we `import "whatwg-fetch"` inside them.

## Why do we need this feature?
Across the test files we have multiple cases where we explicitly import a fetch polyfill (`whatwg-fetch`). The `app.ts` file that actually runs Grafana UI also imports this polyfill. Instead of importing this on a case by case basis, we should instead import it as part of the jest setup, so every single test will run with this, and run in a predictable way based on the real app behaviour.

However - when we start importing this polyfill in the jest `setupTests` file, a number of tests fail, as the polyfill has made the components/hooks work more accurately. This means that some endpoint requests are refused as there is no jest mock or MSW server running to handle them. As we (rightfully) fail when a console.error happens, these tests start failing.

This PR adds an explicit import of `whatwg-fetch` to the files that start failing, and also ensures those tests then start passing again. In a followup PR, I'm going to remove all of the `whatwg-fetch` imports and instead import it once inside `setupTests`.

This means that the tests will then behave more accurately to real behaviour, and we could start moving towards a testing strategy where a MSW server is started before all tests, and the tests don't have to worry about specific API/network behaviour (outside of configuring mock APIs for specific test cases). This means that, for example, a given test may not need to worry about the exact response of the `/api/search` endpoint, but the component tree might render that in some manner.

## Who is this feature for?

Grafana UI developers

### CODE REVIEW

File: public/app/features/playlist/PlaylistNewPage.test.tsx [hunk 1]
```diff
@@ -37,15 +39,11 @@ function getTestContext({ name, interval, items }: Partial<Playlist> = {}) {
 }
 
 describe('PlaylistNewPage', () => {
-  beforeEach(() => {
-    jest.spyOn(console, 'error').mockImplementation(() => {});
```
Comment (tomratcliffe) [inline public/app/features/playlist/PlaylistNewPage.test.tsx:41]: (This is just something else I spotted, we don't need to spy on the console, we can get rid of the error from the test by awaiting the element instead)

File: public/app/features/alerting/unified/RuleEditorGrafanaRules.test.tsx [hunk 2]
```diff
@@ -61,9 +64,13 @@ const mocks = {
   },
 };
 
+const server = setupServer();
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/RuleEditorGrafanaRules.test.tsx:67]: Can we replace some of this logic with `setupMswServer()` which will automatically set up the jest lifecycle hooks too? 

https://github.com/grafana/grafana/blob/3cfe5b5bea0e71296afa07a00e2bde8b4c2a8c82/public/app/features/alerting/unified/mockApi.ts#L412-L430

Comment (tomratcliffe) [reply public/app/features/alerting/unified/RuleEditorGrafanaRules.test.tsx]: Good idea, thanks :+1: I've replaced the uses of it within the alerting code

File: public/app/features/alerting/unified/hooks/useAbilities.test.tsx [hunk 3]
```diff
@@ -137,9 +144,20 @@ describe('alertmanager abilities', () => {
...
 
+    // TODO: Remove server mocking within test once server is run before all tests
+    mockFolderApi(server).folder(
+      (rule.rulerRule as RulerGrafanaRuleDTO).grafana_alert.namespace_uid,
+      mockFolder({
+        accessControl: { [AccessControlAction.AlertingRuleUpdate]: false },
+      })
+    );
+    mockAlertmanagerChoiceResponse(server, defaultAlertmanagerChoiceResponse);
+    server.listen();
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/hooks/useAbilities.test.tsx:159]: Why do we need to call `listen()` in here? `setupMswServer` is calling listen in the `beforeAll`

**Review by soniaAguilarPeiron — APPROVED** [review]
Great job!🙌
Looking forward having MSW server started before all tests!
I added a non blocking comment

File: public/app/features/alerting/unified/hooks/useAbilities.test.tsx (hunk 3 above)
Comment (tomratcliffe) [reply public/app/features/alerting/unified/hooks/useAbilities.test.tsx]: Good spot! Thanks, forgot this one. I've removed it :+1:

**Review by Sergej-Vlasov — APPROVED** [review]

**Review by ifrost — APPROVED** [review]

**Pull request merged**